### PR TITLE
Add a props to control the number of point per pixel threshold

### LINKED ISSUE #191: xExtents does not work in some cases
Hey @rrag, we would love to have your opinion on a bug we are having.

I reproduced it on this branch https://github.com/rrag/react-stockcharts/tree/bugDomain, run `npm run watch` and open `http://localhost:8080/documentation.html#/line_scatter`

 You will see that:

image

note that the first point is 9.11am. the xExtents however start at 9am. 

```
const minDate = new Date(1480579255864); // Thu Dec 01 2016 09:00:55 GMT+0100
const maxDate = new Date(1480580777028); // Thu Dec 01 2016 09:26:17 GMT+0100
xExtents={[minDate, maxDate]}
```

The points before 9.11am are displayed as you can see if you drag the chart:

dec-09-2016 17-26-48

We found it that using `plotFull` props on ChartCanvas will work but is not performant when zooming.

Thanks for your help, it means a lot to us :)

CC @dbrugne

### PR DESCRIPTION
**Pull request by shprink**
Closes #191 

Also use the new logger instead of `console.log`

### CODE REVIEW

File: src/lib/scale/evaluator.js [hunk 1]
```diff
@@ -62,13 +63,11 @@ function extentsWrapper(inputXAccessor, realXAccessor, useWholeData, clamp) {
 	return domain;
 }
 
-function canShowTheseManyPeriods(width, arrayLength) {
-	var threshold = 2; // number of datapoints per 1 px
+function canShowTheseManyPeriods(width, arrayLength, threshold) {
 	return arrayLength > 1 && arrayLength < width * threshold;
```
Comment (iNikNik) [inline src/lib/scale/evaluator.js]: We can just call showMax here:
```javascript
function canShowTheseManyPeriods(width, arrayLength, threshold) {
   return arrayLength > 1 && arrayLength <= showMax(width, threshold)
}
```

File: src/lib/ChartCanvas.jsx [hunk 2]
```diff
@@ -89,7 +89,7 @@ function getXScaleDirection(flipXScale) {
 }
 
 function calculateFullData(props) {
-	var { data: inputData, calculator, plotFull, xScale: xScaleProp, clamp } = props;
+	var { data: inputData, calculator, plotFull, xScale: xScaleProp, clamp, pointsPerPxThreshold = 2 } = props;
```
Comment (iNikNik) [inline src/lib/ChartCanvas.jsx]: Probably we shuould add default `pointsPerPxThreshold` default value to `ChartCanvas.defaultProps`

Comment (shprink) [inline src/lib/ChartCanvas.jsx]: Good point, adding it

File: src/lib/scale/evaluator.js (hunk 1 above)
Comment (shprink) [inline src/lib/scale/evaluator.js]: done ;)

**Pull request merged**